go-common.mk

```diff
@@ -114,9 +108,9 @@ lint:: pre-lint standard-lint post-lint
 pre-lint::
 
 standard-lint::
-	@which golangci-lint > /dev/null 2>&1 || \
-		$(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
-	golangci-lint run $(GOLINTFLAGS)
+	go mod why github.com/golangci/golangci-lint/cmd/golangci-lint 2> /dev/null | grep 'does not need package' > /dev/null 2>&1 && \
+		echo "golangci-lint is not installed in your go module - please add it in a suitable tools.go file. See other go projects for details."
```

Comment:
How many other go projects should one look at?  This feels like it should be a URL which we know has information on how to do this.

Reply:
Well by the time I finish up this ticket, every single go project should be consistent in this, but since I'm holding this for a while anyway, I'll add:

TODO: before merging, make sure this is updated to point to a specific project (this is a bit of a chicken-and-egg problem but not a big deal, I can point the link before merging the other project)